gateway-ha/src/main/java/io/trino/gateway/ha/router/HaGatewayManager.java

```diff
@@ -82,13 +83,33 @@ public Optional<ProxyBackendConfiguration> getBackendByName(String name)
     @Override
     public void deactivateBackend(String backendName)
     {
-        dao.deactivate(backendName);
+        updateClusterActivationStatus(backendName, false, () -> dao.deactivate(backendName));
     }
 
     @Override
     public void activateBackend(String backendName)
     {
-        dao.activate(backendName);
+        updateClusterActivationStatus(backendName, true, () -> dao.activate(backendName));
+    }
+
+    private void updateClusterActivationStatus(String clusterName, boolean newStatus, Runnable action)
```

Comment:
The "action" parameter name is too general. Please consider renaming to `changeActiveStatus` or something.